### DISCUSSION TITLE
How to getSmiles() in js with version 2.4.1?

### DISCUSSION THREAD

**Discussion by dboeglin** [epam/ketcher#1286] [Q&A] [answered]
Hi,

I am trying to export current structure as a SMILES string in js with version 2.4.1.
I was using successfully the following code in with version 2.3.0 (standalone version) for that purpose:

`ketcher.getSmilesAsync(true).then(function(xyz) {console.log(xyz);}`

With version 2.4.1 (standalone), I have the following error: 

'Uncaught TypeError: ketcher.getSmilesAsync is not a function'

Can you let me know how to proceed with version 2.4.1 ?

Thanks

**Comment by AndreiMazol** [comment, marked as answer]
Dear @dboeglin ,

in scope of release 2.4 API of Ketcher has been reworked, to get SMILES you  need to use the following code:

`ketcher.getSmiles().then((res) => console.log(res))`

Regards,
Andrei

**Reply by MichelML** [reply]
this is your answer @dboeglin , please mark this as resolved
